x-pack/plugins/osquery/public/saved_queries/form/use_saved_query_form.tsx

```diff
@@ -20,8 +22,22 @@ interface UseSavedQueryFormProps {
   handleSubmit: (payload: unknown) => Promise<void>;
 }
 
-export const useSavedQueryForm = ({ defaultValue, handleSubmit }: UseSavedQueryFormProps) =>
-  useForm({
+export const useSavedQueryForm = ({ defaultValue, handleSubmit }: UseSavedQueryFormProps) => {
+  const { data } = useSavedQueries({});
```

Comment:
can we move it to the backend?

Reply:
i like the idea of having it fail validation eagerly, but i also like the idea of having redundant protections. i'll add a check in the backend for both this and the scheduled query endpoint

Reply:
i added additional logic to the saved query endpoints, but skipped on the scheduled query endpoints since it seems like they are commented out currently. it seems like all the saved object write logic is done on the frontend anyway, from what i can tell; the saved query ui elements don't actually call into the backend, unless i'm missing something.

Reply:
you're right 🤦 
maybe let's move the validation fix to 7.15 and in this PR only merge telemetry, expired message, and agents count? 
as we're going to introduce new logic for packs in 7.15 maybe we will be able to solve the validation differently

Reply:
i'll rip out the backend logic, but i think it's worthwhile to keep the frontend validation in place to avoid the duplicate id problem. we can always rip it out when we implement something more general purpose.

Reply:
maybe we could move it to 
https://github.com/patrykkopycinski/kibana/blob/feat/cases-lens-markdown-plugin/x-pack/plugins/osquery/public/saved_queries/use_create_saved_query.ts#L40
and
https://github.com/patrykkopycinski/kibana/blob/feat/cases-lens-markdown-plugin/x-pack/plugins/osquery/public/saved_queries/use_update_saved_query.ts#L39

and then we just try to `savedObjectsClient.find({'id': yyyy})` and if results are empty then it means that the `id` is unique, what do you think?

Reply:
it makes a lot of sense to me to put defensive logic around the points where the data is created/modified, so that sounds great. i'll add that.